scene/resources/material.cpp

```diff
@@ -812,6 +821,13 @@ void BaseMaterial3D::_update_shader() {
 			code += "\t}";
 			code += "\tUV /= vec2(h_frames, v_frames);\n";
 			code += "\tUV += vec2(mod(particle_frame, h_frames) / h_frames, floor(particle_frame / h_frames) / v_frames);\n";
+			code += "\tif (particles_anim_blend) {\n";
+			// Store the next frame UV in UV2 for blending.
+			code += "\t\tUV2 /= vec2(h_frames, v_frames);\n";
+			code += "\t\tUV2 += vec2(mod(particle_frame + 1.0, h_frames) / h_frames, floor((particle_frame + 1.0) / h_frames) / v_frames);\n";
```

Comment:
This should use a varying instead of overriding the UV2 builtin. As it is right now, this is going to clash with the UV2 scaling logic below.

You will also need to clamp/mod (as appropriate) ``particle_frame + 1.0`` otherwise it will break towards the end of an animation

Reply:
> You will also need to clamp/mod (as appropriate) particle_frame + 1.0 otherwise it will break towards the end of an animation

Interestingly, i didn't notice any issues from my testing, both when looping is enabled or disabled. That may be due to repeating being enabled in the texture though (it shouldn't require repeat to be enabled).